openfl-tutorials/interactive_api/PyTorch_Histology/envoy/histology_shard_descriptor.py

```diff
@@ -0,0 +1,149 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Histology Shard Descriptor."""
+
+import logging
+import os
+from pathlib import Path
+from typing import Tuple
+from urllib.request import urlretrieve
+from zipfile import ZipFile
+
+import numpy as np
+from PIL import Image
+
+from openfl.interface.interactive_api.shard_descriptor import ShardDataset
+from openfl.interface.interactive_api.shard_descriptor import ShardDescriptor
+from openfl.utilities import tqdm_report_hook
+from openfl.utilities import validate_file_hash
+
+
+logger = logging.getLogger(__name__)
+
+
+class HistologyDataset:
+    """Colorectal Histology Dataset."""
+
+    def __init__(self, data_folder: Path):
+        """Initialize HistologyDataset."""
+        root = Path(data_folder) / 'Kather_texture_2016_image_tiles_5000'
+        classes = [d.name for d in os.scandir(root) if d.is_dir()]
+        class_to_idx = {cls_name: i for i, cls_name in enumerate(classes)}
+        self.samples = []
+        root = root.expanduser()
+        for target_class in sorted(class_to_idx.keys()):
+            class_index = class_to_idx[target_class]
+            target_dir = os.path.join(root, target_class)
+            for root, _, fnames in sorted(os.walk(target_dir, followlinks=True)):
+                for fname in sorted(fnames):
+                    path = os.path.join(root, fname)
+                    item = path, class_index
+                    self.samples.append(item)
+
+    def load_pil(self, path):
+        """Load image."""
+        with open(path, 'rb') as f:
+            img = Image.open(f)
+            return img.convert('RGB')
+
+    def __getitem__(self, index: int):
+        """Get item by index."""
+        path, target = self.samples[index]
+        sample = self.load_pil(path)
+
+        return sample, target
+
+    def __len__(self):
+        """Get length."""
+        return len(self.samples)
+
+
+class HistologyShardDataset(ShardDataset):
```

Comment:
Why don't we just merge the HistologyShardDataset and HistologyDataset classes?

Reply:
Done